core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -561,7 +561,12 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       val consumerRebalanceListener = new InternalRebalanceListenerForNewConsumer(this, customRebalanceListener)
       if (whitelistOpt.isDefined) {
         try {
-          consumer.subscribe(Pattern.compile(whitelistOpt.get), consumerRebalanceListener)
+          val revisedWhitelist = whitelistOpt.get.trim
+            .replace(',', '|')
+            .replace(" ", "")
+            .replaceAll("""^["']+""","")
+            .replaceAll("""["']+$""","") // property files may bring quotes
+          consumer.subscribe(Pattern.compile(revisedWhitelist), consumerRebalanceListener)
```

Comment:
What if we use something like this
```
          val whitelist = Whitelist(whitelistOpt.get)
          consumer.subscribe(Pattern.compile(whitelist.regex), consumerRebalanceListener)
```
so that regex manipulation is hardcoded in one place?